inloop/solutions/views.py

```diff
@@ -99,15 +102,37 @@ def get(self, request, **kwargs):
 
         # TODO: PrettyPrinters should be configurable (for now, we only have one for JUnit)
         result = solution.testresult_set.last()
-        xml_reports = junit.checkeroutput_filter(result.testoutput_set)
-        testsuites = [
-            junit.xml_to_dict(xml) for xml in xml_reports
-        ]
+
+        xml_reports_junit = checkeroutput_filter(result.testoutput_set)
+        testsuites = [xml_to_dict(xml) for xml in xml_reports_junit]
+
+        parser = Parser(solution=solution)
+        checkstyle_context = parser.context(
+            startswith="checkstyle_errors",
+            endswith=".xml",
+            filter_keys=[]
+        )
+
+        checkstyle_files = Parser.extract(data=checkstyle_context["data"], key="file")
+        checkstyle_files = assign_sources_to_files(checkstyle_files, solution)
+        checkstyle_files = assign_code_to_errors(checkstyle_files)
+        checkstyle_files = assign_grouped_errors(checkstyle_files)
+        checkstyle_files = remove_input_path(checkstyle_files)
```

Comment:
This screams for a redesign of the `XMLParser`, which should carry out these 5 steps and return a dict with the three entries `checkstyle_files`, `total_checkstyle_errors`, and `total_checkstyle_warnings`. This would also remove the dodgy re-assigment of the same variable for different purposes.